packages/service-provider-server/src/node/node-cursor.ts

```diff
@@ -377,6 +377,30 @@ class NodeCursor implements Cursor {
     this.cursor.addCursorFlag(flag, true);
     return this;
   }
+
+  async explain(verbosity: string): Promise<any> {
```

Comment:
I think if you want to merge it in as is, I'd definitely add a set of comments as to why `explain` cursor needs to be done this way. I think if we were to forget in two months that explain does not get properly done in the node driver, this function will be a bit perplexing.

Reply:
The weird part i was mentioning is in `explainable_find()` and `explainable_aggregate()`. This part has to stay even after the driver implements explain everywhere :). I can add a comment about the driver returning the full explain, which is intended though, and about the fact we probably should move the adaptations we do here in the mapper and just use the native cursor in this layer since it already implements the interface we need.

FYI. The problem i was talking about, with the node driver, is that `explain` is only implemented for cursors, while we need it for remove / update etc too. Also the output with `aggregate` is incomplete, but it makes no difference here since we have to handle all the cases anyway.